### ISSUE TITLE
DataLine can't open browser from WSL2

### ISSUE THREAD

**Issue by RamiAwar** [RamiAwar/dataline#345]
### Discussed in https://github.com/RamiAwar/dataline/discussions/343

<div type='discussions-op-text'>

<sup>Originally posted by **tlawrence-TAG** October  9, 2024</sup>
pydantic/_internal/_fields.py:200: UserWarning: Field name "api_key" in "ShouldGenerateChartCall" shadows an attribute in parent "OpenAIExtractor[ShouldGenerateChart]"
pydantic/_internal/_fields.py:200: UserWarning: Field name "base_url" in "ShouldGenerateChartCall" shadows an attribute in parent "OpenAIExtractor[ShouldGenerateChart]"
pydantic/_internal/_fields.py:200: UserWarning: Field name "api_key" in "GenerateChartCall" shadows an attribute in parent "OpenAIExtractor[GeneratedChart]"
pydantic/_internal/_fields.py:200: UserWarning: Field name "base_url" in "GenerateChartCall" shadows an attribute in parent "OpenAIExtractor[GeneratedChart]"
pydantic/_internal/_fields.py:200: UserWarning: Field name "api_key" in "ConversationTitleGenerator" shadows an attribute in parent "OpenAICall"
pydantic/_internal/_fields.py:200: UserWarning: Field name "base_url" in "ConversationTitleGenerator" shadows an attribute in parent "OpenAICall"
INFO:     Started server process [13927]
INFO:     Waiting for application startup.
INFO:alembic.runtime.migration:Context impl SQLiteImpl.
INFO:alembic.runtime.migration:Will assume non-transactional DDL.
ERROR:    Traceback (most recent call last):
  File "starlette/routing.py", line 677, in lifespan
  File "contextlib.py", line 204, in __aenter__
  File "dataline/main.py", line 61, in lifespan
**AttributeError: 'NoneType' object has no attribute 'id'**

ERROR:    Application startup failed. Exiting.
/usr/bin/xdg-open: 882: x-www-browser: not found
/usr/bin/xdg-open: 882: firefox: not found
/usr/bin/xdg-open: 882: iceweasel: not found
/usr/bin/xdg-open: 882: seamonkey: not found
/usr/bin/xdg-open: 882: mozilla: not found
/usr/bin/xdg-open: 882: epiphany: not found
/usr/bin/xdg-open: 882: konqueror: not found
/usr/bin/xdg-open: 882: chromium: not found
/usr/bin/xdg-open: 882: chromium-browser: not found
/usr/bin/xdg-open: 882: google-chrome: not found
/usr/bin/xdg-open: 882: www-browser: not found
/usr/bin/xdg-open: 882: links2: not found
/usr/bin/xdg-open: 882: elinks: not found
/usr/bin/xdg-open: 882: links: not found
/usr/bin/xdg-open: 882: lynx: not found
/usr/bin/xdg-open: 882: w3m: not found
xdg-open: no method available for opening 'http://localhost:7377'

I downloaded the linux distro and am running this on WSL2. Any help would be appreciated!</div>

## Solution

Allow failure of web browser opening - then at least it won't crash - user can manually open browser.

Basically add a try catch around webbrowser.open()!

**Comment by anthony2261** [comment]
@RamiAwar `AttributeError: 'NoneType' object has no attribute 'id'` is the issue from posthog when no user is present (fixed in my merged PR - to be released).

**Comment by RamiAwar** [comment]
Fixed in latest release.
